### ISSUE TITLE
[a11y]: NumberInput - Screen Reader Not Announcing Incremented/Decremented Values

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16529]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

When the keyboard is used to increment or decrement the values via the arrow keys, the screen reader does not announce the updated value.
If the screen reader does not announce the value change in the input field, it’s difficult for users to understand it on their own. This defect mainly impacts users who are blind. 

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-numberinput--default

### Recommend Fix
Ensure JAWS announce the incremented and decremented values accordingly when they are changed.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
I'm not 100% sure but this might be partially solved when `type="text"`, https://github.com/carbon-design-system/carbon/pull/19471

**Comment by wkeese** [comment]
Does JAWS announce for the example on https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/number?

**Comment by heloiselui** [comment]
This issue was retested by your a11y team and is now fixed. I’m going to close it because of that, but if the issue comes back, we can open a new one or reopen this one.

<img width="1441" height="811" alt="Image" src="https://github.com/user-attachments/assets/a0731d1e-4f5a-4cdd-a7d5-a331771c4313" />
